notebook/agentchat_hierarchy_flow_using_select_speaker.ipynb

```diff
@@ -0,0 +1,370 @@
+      "--------------------------------------------------------------------------------\n",
+      "\u001b[33mB1\u001b[0m (to chat_manager):\n",
+      "\n",
+      "NEXT: A1\n",
+      "Hello A1, Could you please ask A2 and A3 to provide the values for x and y so we can compute their product?\n",
+      "\n",
+      "--------------------------------------------------------------------------------\n",
+      "\u001b[33mA1\u001b[0m (to chat_manager):\n",
+      "\n",
+      "Sure, I'll ask them for the values. Hey A2, A3! Can you give us the values of x and y for computation?\n",
+      "\n",
+      "--------------------------------------------------------------------------------\n",
+      "\u001b[33mA2\u001b[0m (to chat_manager):\n",
+      "\n",
+      "Certainly, A1. The value of x is 9. However, I do not know the value of y. You'll have to ask A3 for that.\n",
+      "\n",
+      "--------------------------------------------------------------------------------\n",
+      "\u001b[33mA3\u001b[0m (to chat_manager):\n",
+      "\n",
+      "Alright! since I know the value of y, I can tell that y = 5.\n",
+      "\n",
+      "--------------------------------------------------------------------------------\n",
+      "\u001b[33mA1\u001b[0m (to chat_manager):\n",
+      "\n",
+      "Great, based on what A2 and A3 have provided, the value of x is 9 and y is 5. So, the product of x and y is 45.\n",
+      "\n",
+      "--------------------------------------------------------------------------------\n",
+      "\u001b[33mB1\u001b[0m (to chat_manager):\n",
+      "\n",
+      "Thank you, A1. So B2, the product of x and y is 45.\n",
+      "\n",
+      "--------------------------------------------------------------------------------\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Initialization\n",
+    "agents_A = [\n",
+    "    AssistantAgent(name='A1', system_message=\"You are a team leader A1, your team consists of A2, A3. You can talk to the other team leader B1, whose team member is B2.\"),\n",
+    "    AssistantAgent(name='A2', system_message=\"You are team member A2, you know the secret value of x but not y, x = 9. Tell others x to cooperate.\"),\n",
+    "    AssistantAgent(name='A3', system_message=\"You are team member A3, You know the secret value of y but not x, y = 5. Tell others y to cooperate.\"),\n",
+    "]\n",
+    "\n",
+    "agents_B = [\n",
+    "    AssistantAgent(name='B1', system_message=\"You are a team leader B1, your team consists of B2. You can talk to the other team leader A1, whose team member is A2, A3. Use NEXT: A1 to suggest talking to A1.\"),\n",
+    "    AssistantAgent(name='B2', system_message=\"You are team member B2. Your task is to find out the value of x and y and compute the product. Once you have the answer, say out the answer and append a new line with TERMINATE.\")\n",
+    "]\n",
+    "\n",
+    "# Terminates the conversation when TERMINATE is detected.\n",
+    "user_proxy = autogen.UserProxyAgent(\n",
```

Comment:
there is a system message, execution is turned off, and human input is never. But there is no llm_config